docs/pages/blog/hello-grid-version-2.md

```diff
@@ -0,0 +1,144 @@
+```
+
+## The problems and motivation
+
+- To fix the migration pain points ([#26266](https://github.com/mui/material-ui/issues/29266) and [#31244](https://github.com/mui/material-ui/issues/31244)) for the grid component which is caused by the [implementation changes](https://github.com/mui/material-ui/pull/24332) introduced in v5.
+- It is an opportunity to add the long-awaited [offset feature](https://github.com/mui/material-ui/issues/11251) to the grid component.
+- The grid container is using child combinator (`>`) to style the grid items which makes [customization](https://github.com/mui/material-ui/issues/28855) harder.
+- The [limitation](https://mui.com/material-ui/react-grid/#nested-grid) on nested grids could be fixed.
+- Unnecessary `item` prop.
+
+## What's new
+
+The `Grid v2` has been rewritten from scratch with CSS variables and `calc()` which lets us fix the issues and apply new features easily.
+
+### 💥 Breaking change
+
+The negative margin in `Grid v2` spreads equally on all sides by default (same as the `Grid` in Material UI v4).
+
+:::info
+We believe that the migration from Grid `v1` to `v2` will be smooth for most of the cases.
+:::
+
+### 🚀 New features
+
+#### [Preventing scrollbar](/material-ui/react-grid2/#disable-the-scrollbar)
+
+A new prop called `disableEqualOverflow` is added to properly prevent the scrollbar issue in small viewports.
+
+#### [Offset](/material-ui/react-grid2/#offset)
+
+We have added the long-awaited offset feature to `Grid v2`. Thanks to CSS variables, it lets us implement the feature with just a few lines of code.
+
+### ✨ Improvements
+
+#### Simplifying logic
+
+Some parts of the calculation are delegated to CSS by using CSS variables and `calc()` together. They reduces the need of javascript and makes the component scalable.
+
+For example, the `Grid v1` currently calculates the width by reading the value from a React context. However, with CSS variables in place, we can remove the React context and hand over the work to CSS instead:
+
+```js
+{
+  // --Grid-columns is defined in the grid container
+  width: `calc(100% * ${value} / var(--Grid-columns))`,
+}
+```
+
+Another example is how the padding is calculated in grid items:
+
+In `Grid v1`, grid items has no information about the padding it should have because the `spacing` prop is defined on the grid container. So, the simplest way is to control the padding from the grid container using CSS child combinator (`>`):
+
+```js
+function getOffset(val) {
+  const parse = parseFloat(val);
+  return `${parse}${String(val).replace(String(parse), '') || 'px'}`;
+}
+// ...The code is shorten for readability
+({
+  [`& > .${gridClasses.item}`]: {
+    // Grid item's padding is controlled by its container
+    paddingTop: getOffset(themeSpacing),
+  },
+});
+```
+
+However, in `Grid v2`, the javascript logic can be separated independently because CSS variables shift the communication between container and item to CSS.
```

Comment:
```suggestion
In v2, the JavaScript logic is independent from the CSS, because CSS variables shift the communication between container and item over to the CSS.
```

Reply:
> the JavaScript logic is independent from the CSS

This phrase has a different meaning from what I intended to communicate.

Or should it be
```suggestion
In v2, the JavaScript logic is independent between grid container and item, because CSS variables shift the communication over to the CSS.
```

Reply:
Ohhh I see what you're saying, something like this maybe?

```suggestion
In v2, the communication between the grid container and its items happens via CSS instead of JavaScript, so that the logic behind these components can be better separated.
```